src/lib/notification_service/formatter_provider.py

```diff
@@ -0,0 +1,24 @@
+from .formatter import HtmlFormatter
+
+
+class FormatterProvider:
+    """Formatter factory."""
+
+    def __init__(self):
+        self._formatters = {}
+
+    def register_sender(self, delivery_method, sender):
+        self._formatters[delivery_method] = sender
+
+    def get(self, delivery_method):
+        formatter = self._formatters.get(delivery_method)
+
+        if not formatter:
+            raise ValueError(f"Delivery method {delivery_method} is not supported.")
+
+        return formatter()
+
+
+formatters = FormatterProvider()
+formatters.register_sender("AWS_SES", HtmlFormatter)
+formatters.register_sender("SMTP", HtmlFormatter)
```

Comment:
Just though if it possible to have just one provider for both Sender and Formatter? Like 
```
provider = NotificationProvider()
provider.register_sender("AWS_SES", create_aws_ses_sender, HtmlFormatter)
```
It is just an idea since we have both formatter and sender per each delivery method, but not sure if it possible in such case :)
Everything else looks great, as usual!

Reply:
Thanks for the comment, but I believe it is better not to mix these 2 different entities, because get method and register_sender should retrieve and update different dictionaries.